### PR TITLE
feat: Add support for Gatling Enterprise Component License

### PR DESCRIPTION
**Pull request by guilgaly**
Motivation:

We are introducing components which are:

- published separately from Gatling OSS
- intended for use with Gatling Enterprise
- published (in a limited version) to Maven Central for user convenience, under a specific license

For instance, the Gatling gRPC component.

Modifications:

- Add another type of license file to include in JARs (in addition to Apache v2)
- Move sonatype publication settings into a dedicated plugin, instead of the GatlingOssPlugin, so that they can be reused
- introduce the GatlingEnterpriseComponentPlugin to be used to build the new components

Result:

- No change in use of GatlingCorpPlugin or GatlingOssPlugin
- GatlingEnterpriseComponentPlugin can now be used for the new type of components

### CODE REVIEW

**Review by slandelle — APPROVED** [review]

File: src/main/scala/io/gatling/build/license/GatlingLicenseFileCommon.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+private[license] object GatlingLicenseFileCommon {
+  import GatlingLicenseFileKeys._
+
+  def projectSettings: Seq[Def.Setting[_]] = Seq(
+    Compile / gatlingLicenseFileTask := {
+      val file = (Compile / resourceManaged).value / "META-INF" / "LICENSE"
+      val license = (Compile / gatlingLicenseFile).value
+      val contents = IO.readStream(getClass.getClassLoader.getResourceAsStream(license), Charset.forName("UTF-8"))
+      IO.write(file, contents)
```
Comment (Isammoc) [inline src/main/scala/io/gatling/build/license/GatlingLicenseFileCommon.scala:32]: `IO.transfer`?

File: src/main/scala/io/gatling/build/sonatype/GatlingSonatypePlugin.scala [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  override def requires: Plugins =
+    GatlingBasicInfoPlugin &&
+      GatlingReleasePlugin &&
+      GatlingPublishPlugin &&
+      Sonatype
+
+  trait GatlingSonatypeKeys {
+    val gatlingPublishToSonatype = settingKey[Boolean]("true to publish to sonatype")
+  }
+  object GatlingSonatypeKeys extends GatlingSonatypeKeys
```
Comment (Isammoc) [inline src/main/scala/io/gatling/build/sonatype/GatlingSonatypePlugin.scala:44]: redundant name?
Either as an first citizen, or rename it
```suggestion
  trait Keys {
    val gatlingPublishToSonatype = settingKey[Boolean]("true to publish to sonatype")
  }
  object Keys extends Keys
```

Comment (guilgaly) [reply src/main/scala/io/gatling/build/sonatype/GatlingSonatypePlugin.scala:44]: This was to follow the same convention as the other existing plugins (e.g. `GatlingBasicInfoPlugin.GatlingBasicInfoKeys`, `GatlingPublishPlugin.GatlingPublishKeys`, etc.)

Comment (Isammoc) [reply src/main/scala/io/gatling/build/sonatype/GatlingSonatypePlugin.scala:44]: Oh! Ok :disappointed: 

Maybe we should have our own convention :shrug: 

But let's keep it for now

File: src/main/scala/io/gatling/build/sonatype/GatlingSonatypePlugin.scala [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        startStateWithSonatypeConf
+      )
+    }
+
+    state.log.info("Closing sonatype staging")
+    Def.unit(releaseStepCommandAndRemaining("sonatypeClose")(sonatypeOpenState))
+    endState
+  }
+
+  def ensurePublishableVersion(str: String): Boolean = str.matches("""\d+\.\d+\.\d+(\.\d+)?(-M\d+)?""")
```
Comment (guilgaly) [inline src/main/scala/io/gatling/build/sonatype/GatlingSonatypePlugin.scala:117]: we now accept an optional 4th component of the version number (can be used by components like gatling-grpc, same principle we already use for the Gradle plugin versions).

**Pull request merged**